### PR TITLE
remove test-data-fixture from rest of the API tests

### LINKED ISSUE #2298: Test data creation
We could improve the test data creation more. Ideas:

- remove dependency to any example test data by having the tests explicitly call test helpers
- make helpers more convenient for test data cases (i.e. create org if it does not exist so you don't have to create it manually)
- remove test-data-fixture (after all tests create their own data)
- see also https://github.com/CSCfi/rems/pull/2294#issuecomment-686922170
https://github.com/CSCfi/rems/issues/2297
https://github.com/CSCfi/rems/issues/2334

- [x] test-data-fixture direct references removed from tests

- [x] test-data-fixture removed from standalone-fixture

- [x] test-data-fixture removal from api-fixture
--> namespaces to be fixed:

- [x] bin/kaocha  --focus 'rems.api.test-api/data-exception-test'
- [x] bin/kaocha  --focus 'rems.api.test-api/test-health-api'
- [x] bin/kaocha  --focus 'rems.api.test-api/test-api-key-security'
- [x] bin/kaocha  --focus 'rems.test-redirects/test-attachment-download'
- [x] bin/kaocha  --focus 'rems.test-redirects/test-license-attachment-download'
- [x] bin/kaocha  --focus 'rems.api.test-catalogue-items/change-form-test'
- [x] bin/kaocha  --focus 'rems.api.test-catalogue-items/test-enable-archive'
- [x] bin/kaocha  --focus 'rems.api.test-catalogue-items/catalogue-items-api-test'
- [x] bin/kaocha  --focus 'rems.api.test-catalogue-items/catalogue-items-edit-test'
- [x] bin/kaocha  --focus 'rems.api.test-audit-log/test-audit-log'
- [x] bin/kaocha  --focus 'rems.api.test-organizations/organization-api-status-test'
- [x] bin/kaocha  --focus 'rems.api.test-organizations/organization-duplicate-key-test'
- [x] bin/kaocha  --focus 'rems.api.test-organizations/organizations-api-security-test'
- [x] bin/kaocha  --focus 'rems.api.test-organizations/organizations-api-test'
- [x] bin/kaocha  --focus 'rems.api.test-workflows/workflows-enabled-archived-test'
- [x] bin/kaocha  --focus 'rems.api.test-workflows/workflows-api-test'
- [x] bin/kaocha  --focus 'rems.api.test-workflows/workflows-edit-test'
- [x] bin/kaocha  --focus 'rems.api.test-workflows/workflows-api-filtering-test'
- [x] bin/kaocha  --focus 'rems.api.test-workflows/workflows-api-security-test'
- [x] bin/kaocha  --focus 'rems.api.test-email/test-send-reminders'
- [x] bin/kaocha  --focus 'rems.api.test-email/test-send-handler-reminder'
- [x] bin/kaocha  --focus 'rems.api.test-email/test-send-reviewer-reminder'
- [x] bin/kaocha  --focus 'rems.api.test-over-http/test-allocate-external-id'
- [x] bin/kaocha  --focus 'rems.api.test-over-http/test-api-sql-timeouts'
- [x] bin/kaocha  --focus 'rems.application.test-search/test-application-search'
- [x] bin/kaocha  --focus 'rems.api.test-blacklist/test-blacklist'
- [x] bin/kaocha  --focus 'rems.api.test-end-to-end/test-end-to-end'
- [x] bin/kaocha  --focus 'rems.api.test-end-to-end/test-blacklisting'
- [x] bin/kaocha  --focus 'rems.api.test-end-to-end/test-approver-rejecter-bots'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-attachment-zip'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-validation'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-comment-attachments'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-close'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-commands'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-revoke'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-submit'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-api-attachments'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-api-license-attachments'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-api-session'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-create'
- [x] bin/kaocha  --focus 'rems.api.test-applications/pdf-smoke-test'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-todos'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-raw'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-listing'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-hiding-sensitive-information'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-approve-with-end'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-delete'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-applications-api-security'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-application-export'
- [x] bin/kaocha  --focus 'rems.api.test-applications/test-decider-workflow'
- [x] bin/kaocha  --focus 'rems.test-event-notification/test-event-notification'
- [x] bin/kaocha  --focus 'rems.api.test-licenses/licenses-api-enable-archive-test'
- [x] bin/kaocha  --focus 'rems.api.test-licenses/licenses-api-test'
- [x] bin/kaocha  --focus 'rems.api.test-licenses/licenses-api-filtering-test'
- [x] bin/kaocha  --focus 'rems.api.test-licenses/licenses-api-security-test'
- [x] bin/kaocha  --focus 'rems.api.test-permissions/permissions-test-content'
- [x] bin/kaocha  --focus 'rems.api.test-permissions/permissions-test-api-disabled'
- [x] bin/kaocha  --focus 'rems.api.test-permissions/permissions-test-security'
- [x] bin/kaocha  --focus 'rems.api.test-resources/resources-api-create-test'
- [x] bin/kaocha  --focus 'rems.api.test-resources/resources-api-enable-archive-test'
- [x] bin/kaocha  --focus 'rems.api.test-resources/resources-api-filtering-test'
- [x] bin/kaocha  --focus 'rems.api.test-resources/resources-api-security-test'
- [x] bin/kaocha  --focus 'rems.api.test-users/active-api-test'
- [x] bin/kaocha  --focus 'rems.api.test-users/users-api-test'
- [x] bin/kaocha  --focus 'rems.api.test-users/users-api-security-test'
- [x] bin/kaocha  --focus 'rems.api.test-catalogue/catalogue-api-test'
- [x] bin/kaocha  --focus 'rems.api.test-catalogue/catalogue-api-security-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/forms-api-security-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/forms-api-filtering-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/forms-api-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/forms-api-visible-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/form-enabled-archived-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/form-edit-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/form-editable-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/forms-api-privacy-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/test-form-missing-languages'
- [x] bin/kaocha  --focus 'rems.api.test-forms/option-form-item-test'
- [x] bin/kaocha  --focus 'rems.api.test-forms/forms-api-all-field-types-test'
- [x] bin/kaocha  --focus 'rems.api.test-administration/test-archiving-disabling'
- [x] bin/kaocha  --focus 'rems.api.test-user-settings/user-settings-api-test'

### PR DESCRIPTION
**Pull request by opqdonut**
Now api-fixture creates no data, instead the test namespaces
explicitly create their test data.

Closes #2298

# Checklist for author

Remove items that aren't applicable, check items that are done.

## Reviewability
- [x] Link to issue

## Testing
- [x] Complex logic is unit tested
- [x] Valuable features are integration / browser / acceptance tested automatically

### CODE REVIEW

File: test/clj/rems/api/test_administration.clj [hunk 1]
```diff
@@ -1,90 +1,88 @@
...
+            [rems.db.testing :refer [owners-fixture +test-api-key+]]))
 
 (use-fixtures
   :once
-  api-fixture)
+  api-fixture-without-data
+  owners-fixture)
 
-(def api-key "42")
 (def user-id "owner")
```
Comment (Macroz) [inline test/clj/rems/api/test_administration.clj:13]: Could we just inline this? I think it's merely confusing us as to who does things (user vs. owner).

File: test/clj/rems/api/test_email.clj [hunk 2]
```diff
@@ -4,14 +4,14 @@
...
             [ring.mock.request :refer :all]))
 
 (use-fixtures
-  :once
-  api-fixture)
+  :each
+  api-fixture-without-data)
 
-(def api-key "42")
 (def user-id "developer")
```
Comment (Macroz) [inline test/clj/rems/api/test_email.clj:15]: Same here, the constant does not seem useful to us.

File: test/clj/rems/api/test_administration.clj [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
 (use-fixtures
   :once
-  api-fixture-without-data
+  api-fixture
```
Comment (Macroz) [inline test/clj/rems/api/test_administration.clj]: :clinking_glasses:

**Review by Macroz — APPROVED** [review]
Great job! 

I only noticed that we have some not so useful var declarations in a few places. Can fix here or later.

File: test/clj/rems/api/test_administration.clj (hunk 1 above)
Comment (opqdonut) [reply test/clj/rems/api/test_administration.clj:13]: sure

File: test/clj/rems/api/test_email.clj (hunk 2 above)
Comment (opqdonut) [reply test/clj/rems/api/test_email.clj:15]: done

**Pull request merged**